### PR TITLE
Kelsonic 17779 r s escape descriptions

### PR DESCRIPTION
**Pull request by kelsonic**
## Description
**Issue:** https://github.com/department-of-veterans-affairs/va.gov-team/issues/17779

This PR strips HTML, strips newlines, and unescapes article descriptions for R&S search.

Introducing a well-known library for decoding called https://github.com/mathiasbynens/he since we don't have access to `document.createElement` or `new DOMParser()` during the build and because neither liquid nor lodash had sufficient unescape methods.

## Testing done
Locally

### Before

localhost_3001_resources_search__query=my+healthevet (1)

### After

localhost_3001_resources_search__query=my+healthevet

## Acceptance criteria
- [x] strips HTML, strips newlines, and unescapes article descriptions for R&S search

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by timwright12 — APPROVED** [review]

File: src/site/stages/build/plugins/create-resources-and-support-section.js [hunk 1]
```diff
@@ -291,10 +292,13 @@ function createArticleListingsPages(files) {
...
-  const sanitized = liquid.filters.strip_html(text);
-  const truncated = liquid.filters.truncate(sanitized, 200);
+  const sanitizedText = liquid.filters.strip_html(text);
+  const strippedNewlines = liquid.filters.strip_newlines(sanitizedText);
+  const decodedText = he.decode(strippedNewlines);
 
-  if (sanitized !== truncated) {
+  const truncated = liquid.filters.truncate(decodedText, 190);
+
+  if (decodedText !== truncated) {
```
Comment (ncksllvn) [inline src/site/stages/build/plugins/create-resources-and-support-section.js:301]: We could leverage the bug in `Cheerio.load` to decode these entities 😂 

I think I would have done something to ensure that if the truncation occurs in the middle of an entity, that we extend that truncation's slice index to the HTML entity's `;` so HTML entity doesn't get cut off. This makes sense to me but it's a little disappointing to see a new package required for this single line. 

How come `lodash/unescape` didn't do the job?

File: package.json [hunk 2]
```diff
@@ -307,6 +307,7 @@
     "fast-levenshtein": "^2.0.6",
     "fine-uploader": "^5.16.2",
     "foundation-sites": "^5.5.3",
+    "he": "^1.2.0",
```
Comment (ncksllvn) [inline package.json:310]: The author of this library really didn't put a lot into naming this one lol. I think we need to run `yarn install` to after this though

File: src/site/stages/build/plugins/create-resources-and-support-section.js (hunk 1 above)
Comment (kelsonic) [reply src/site/stages/build/plugins/create-resources-and-support-section.js:301]: Right? So `lodash/unescape` you can read [here](https://lodash.com/docs/4.17.15#unescape) only escapes the HTML entities &amp;, &lt;, &gt;, &quot;, and &#39;, but we actually use many more than that within those descriptions.

Comment (kelsonic) [reply src/site/stages/build/plugins/create-resources-and-support-section.js:301]: Lodash then recommends using `he`, which is how I found that lib

File: package.json (hunk 2 above)
Comment (kelsonic) [reply package.json:310]: Ah for the yarn.lock you mean? Good idea

File: src/site/stages/build/plugins/create-resources-and-support-section.js (hunk 1 above)
Comment (ncksllvn) [reply src/site/stages/build/plugins/create-resources-and-support-section.js:301]: Ah okay! Thanks

**Review by ncksllvn — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (kelsonic) [reply package.json:310]: Hmm, so I ran `yarn install` and it's not showing up in the `yarn.lock`. So then I deleted the `yarn.lock` and ran it, still no sign of an updated `he` within it. Very odd

Comment (timwright12) [reply package.json:310]: @kelsonic I can see it in mine when I pull the branch. Searching for `"he` didn't work, but `he@` does so I think you're good:

Screen Shot 2021-03-18 at 10 20 33 AM

Comment (kelsonic) [reply package.json:310]: awesome, thanks!

**Pull request merged**